File: Java/src/main/java/behavioral/observer/messaging/PhoneClient.java
package behavioral.observer.messaging;

public class PhoneClient extends Observer{

    private final Subject subject;

    public PhoneClient(Subject subject) {
        this.subject = subject;
        subject.attach(this);
    }

    public void addMessage(String message) {
        subject.setState(message + " - sent from phone");
    }

    @Override
    void update() {
       System.out.println("Phone stream: " + subject.getState());
    }
}

File: C#/Behavioral/Observer/DesignPatterns.BryanHansen.Observer/PhoneClient.cs
using System;

namespace DesignPatterns.BryanHansen.Observer
{
    public class PhoneClient : Observer
    {
        private readonly Subject _subject;

        public PhoneClient(Subject subject)
        {
            _subject = subject;
            subject.Attach(this);
        }

        public void AddMessage(string message)
        {
            _subject.SetState($"{message} - sent from phone");
        }

        public override void Update()
        {
            Console.WriteLine($"Phone stream: {_subject.GetState()}");
        }
    }
}